src/colorizer/utils/data_utils.ts

```diff
@@ -121,3 +135,11 @@ export function getInRangeLUT(dataset: Dataset, thresholds: FeatureThreshold[]):
   }
   return inRangeIds;
 }
+
+/**
+ * Sanitizes a string name to a key for internal use. Replaces all non-alphanumeric characters with underscores,
+ * and converts the string to lowercase.
+ */
+export function getKeyFromName(name: string): string {
+  return name.toLowerCase().replaceAll(/[^a-z0-9_]/g, "_");
```

Comment:
Is there any chance that this method of generating a key would result in identical keys for different features? Or is that even a thing we need to be concerned about? Like what would happen if you ended up with two different features with the same key?

Reply:
Yup, definitely a concern. I'm adding a validation step in the dataset writing tools right now that checks for this. If this does happen, the second feature would overwrite the first in the map that tracks all the features, and the first would essentially "disappear" from the UI.

Reply:
Edit to add: I can likely add in an extra warning that will print to the console for any duplicated keys.

Reply:
I'll leave that up to you! You have a better feeling for the users of this app and how likely it is for duplicate keys to emerge, but I imagine having a console warning could be helpful for debugging in the future, just in case (if it's not too hard to add)